pyxem/utils/indexation_utils.py

```diff
@@ -603,16 +555,42 @@ def _match_polar_to_library_cpu(
     N is the number of templates and R the number of spots in the template
     with the maximum number of spots
     """
-    correlations = _correlate_polar_to_library_cpu(
-        polar_image, r_templates, theta_templates, intensities_templates
-    )
-    correlations_mirror = _correlate_polar_to_library_cpu(
-        polar_image,
-        r_templates,
-        (polar_image.shape[0] - theta_templates) % polar_image.shape[0],
-        intensities_templates,
+    N = r_templates.shape[0]
+    D = r_templates.shape[1]
+    n_shifts = polar_image.shape[0]
+    best_in_plane_shift = np.empty(N, dtype=np.int32)
+    best_in_plane_shift_m = np.empty(N, dtype=np.int32)
+    best_in_plane_corr = np.empty(N, dtype=polar_image.dtype)
+    best_in_plane_corr_m = np.empty(N, dtype=polar_image.dtype)
+
+    for template in prange(N):
+        inplane_cor = np.zeros(n_shifts)
+        inplane_cor_m = np.zeros(n_shifts)
+        for spot in range(D):
+            rsp = r_templates[template, spot]
+            if rsp == 0:
+                break
+            tsp = theta_templates[template, spot]
+            isp = intensities_templates[template, spot]
+            spl = n_shifts - tsp
```

Comment:
Could you give this variable a better name?

Reply:
I called it "split" for now. This is really the key to the improved speed: using array slices instead of atomic operations calculating individual indices from the arrays as in the previous implementation. The simultaneous calculation of the normal and mirrored templates did not substantially improve speed, maybe 10%.